### PR TITLE
Run R package checks via tic and error only on warnings

### LINKED ISSUE #764: Fail builds where R CMD has NOTEs
e.g. https://travis-ci.org/github/cwida/duckdb/builds/708595115#L713 from #757 has 3 NOTEs but is successful.

For this, {tic} will come in very handy. Need to find a way to integrate with existing infrastructure (e.g. S3 upload).

Let's merge #762 before tackling this.

### PR DESCRIPTION
**Pull request by pat-s**
fixes #764 

Also

- Linux and Mac runners use a package cache now
- A code coverage is run in the `after_success` stage

### CODE REVIEW

File: tools/rpkg/build-linux [hunk 1]
```diff
@@ -5,9 +5,11 @@ set -x
...
 R CMD build .
-R CMD INSTALL duckdb_*.tar.gz
-R CMD check --as-cran duckdb_*.tar.gz
+R -q -e 'tic::script()'
 
 R -q -e "tools::write_PACKAGES(dir = '.', type = 'source')"
+
+python ../upload-s3.py rstats/src/contrib duckdb_*.tar.gz PACKAGES*
+python ../upload-s3.py rstats duckdb_*.tar.gz
+python3 ../../scripts/asset-upload.py duckdb_r_src.tar.gz=duckdb_*.tar.gz
```
Comment (krlmlr) [inline tools/rpkg/build-linux:15]: It seems inconsistent to have this code in `build-*` but only on Mac and Linux. Should this remain the responsibility of `.travis.yml`, perhaps as part of a separate script?

**Review by krlmlr — COMMENTED** [review]
This will actually give failing builds when `R CMD check` fails.

**Comment by hannesmuehleisen** [comment]
I see 
`Non-standard files/directories found at top level:
576  ‘deploy-linux’ ‘deploy-mac’ ‘deploy-win’ ‘duckdb_0.2.0.tgz’
577`
Do those need to be added to the build ignore file or does tic just build in the src dir?

**Comment by hannesmuehleisen** [comment]
Looking good, I am just wondering about these rather big blobs in before_install. Is that something verbatim from tic?

**Comment by pat-s** [comment]
We add `ccache` to speed up source installations of R package deps. For duckdb it also has the effect that it would decrease the internal compiling time when building the package.
I just see that the `ccache` part is not cached currently so it is not used.

Let me do a quick check to confirm it would actually do its job if I would enable it correctly.

**Comment by hannesmuehleisen** [comment]
Looks pretty good no?

**Comment by pat-s** [comment]
Yes, on Linux we go down from 12 to 5 mins, on mac from 24 to 12 mins (compared to runtime in master).

No ccache on Windows.

Ready from my side.

**Pull request merged**